src/backend/pagemap.h

```diff
@@ -223,6 +238,21 @@ namespace snmalloc
       return body[p >> SHIFT];
     }
 
+    /**
+     * Return the starting address corresponding to a given entry within the
+     * Pagemap. Also checks that the reference actually points to a valid entry.
+     */
+    [[nodiscard]] address_t get_address(const T& t) const
+    {
+      address_t entry_offset = address_cast(&t) - address_cast(body);
+      address_t entry_index = entry_offset / sizeof(T);
+      if (entry_offset % sizeof(T) != 0 || entry_index >= num_entries())
```

Comment:
Should this be an assert?  Is this code likely to be performance sensitive?

Reply:
While I don't expect this code to be performance sensitive (not more than Pagemap::get), I think that an assert might be more appropriate, since it only does address computation and not pointer indexing.